docs/testplan/transceiver_onboarding_test_plan.md

```diff
@@ -0,0 +1,436 @@
+
+    ```yaml
+    topology:
+        <device_name>:
+            <port_name>:
+                active_firmware: <active_firmware_version>
+                inactive_firmware: <inactive_firmware_version>
+                cmis_rev: <cmis_revision>
+                vendor_date: <vendor_date_code>
+                vendor_name: <vendor_name>
+                vendor_oui: <vendor_oui>
+                vendor_pn: <part_number>
+                vendor_rev: <revision_number>
+                vendor_sn: <serial_number>
+                dual_bank_support: <yes_or_no>
+    ```
+
+#### 1.1 Link related tests
+
+The following tests aim to validate the link status and stability of transceivers under various conditions.
+
+| Step | Goal | Expected Results |
+|------|------|------------------|
+| Issue CLI command to shutdown a port | Validate link status using CLI configuration | Ensure that the link goes down |
+| Issue CLI command to startup a port | Validate link status using CLI configuration | Ensure that the link is up and the port appears in the LLDP table. |
+| In a loop, issue startup/shutdown command 100 times | Stress test for link status validation | Ensure link status toggles to up/down appropriately with each startup/shutdown command. Verify ports appear in the LLDP table when the link is up |
+| Restart `xcvrd` | Test link and xcvrd stability | Confirm `xcvrd` restarts successfully without causing link flaps for the corresponding ports, and verify their presence in the LLDP table. Also ensure that xcvrd is up for at least 2 mins |
+| Induce I2C errors and restart `xcvrd` | Test link stability in case of `xcvrd` restart + I2C errors | Confirm `xcvrd` restarts successfully without causing link flaps for the corresponding ports, and verify their presence in the LLDP table |
+| Restart `pmon` | Test link stability | Confirm `xcvrd` restarts successfully without causing link flaps for the corresponding ports, and verify their presence in the LLDP table |
+| Restart `swss` | Validate transceiver re-initialization and link status post container restart | Ensure `xcvrd` restarts and the expected ports link up again, with port details visible in the LLDP table |
+| Restart `syncd` | Validate transceiver re-initialization and link status post container restart | Ensure `xcvrd` restarts and the expected ports link up again, with port details visible in the LLDP table |
+| Perform a config reload | Test transceiver re-initialization and link status | Ensure `xcvrd` restarts and the expected ports link up again, with port details visible in the LLDP table |
+| Execute a cold reboot | Validate transceiver re-initialization and link status post-device reboot | Confirm the expected ports link up again post-reboot, with port details visible in the LLDP table |
+| Execute a warm reboot | Test link stability through the warm reboot | Ensure `xcvrd` restarts and maintains link stability for the interested ports, with their presence confirmed in the LLDP table |
```

Comment:
Do we need a new test to check the link status after a firmware update?

Reply:
@vvolam I think having the link status check as part of firmware update will ensure link stability is verified based on the relevant scenario being tested.